# 22 March 2018
# A. Vershinina
# Goal: compute a genome coverage at a desired quantile. Useful for DP filtering of VCF files.
# In: genome coverage table generated by bedtools. 
# Bedtools command should be something like:
# bedtools genomecov -ibam genome.bam -g genome.info > genome.cov.tab
# Out: genome.quantile.txt 
# This file will contain the lowest coverage in a tie. 
# Inspired by https://stackoverflow.com/questions/32000434/r-quantile-on-frequency-values

args = commandArgs(trailingOnly=TRUE)
if (length(args) == 0){
  cat("Syntax: Rscript catch_quantile.R [ genome.cov.tab] [quantile]\n")
  cat("Example: Rscript catch_quantile.R  genome.cov.tab 0.95 \n")
  quit()
}

library(dplyr)

#~~~~~~~Load the data 
df <- read.csv(args[1], sep = "\t", header=F)
df_genome <- df[df$V1=="genome",]
cov <- as.numeric(df_genome$V2)
frequency <- as.numeric(df_genome$V3)
dta <- data.frame(cov, frequency)
qtle <- args[2]

#~~~~~~~~Compute cumulative frequency
dta$cumfreq <- cumsum(dta$frequency)/sum(dta$frequency)

#~~~~~~~~Get coverage from desired quantile
q <- (as.numeric(dta$cov[dta$cumfreq >= qtle][1]))

#~~~~~~~~Create output with a correct name
n <- basename(args[1])
fname <- substr(n, 1, nchar(n)-4)
cat(q, file=paste(fname, qtle, "q.txt", sep="."), sep="")
